cpu/cortexm_common/Makefile.dep

```diff
@@ -4,7 +4,7 @@ USEMODULE += cortexm_common
 # include common periph code
 USEMODULE += cortexm_common_periph
 
-ifneq (,$(filter picolibc,$(FEATURES_USED)))
+ifneq (,$(filter picolibc,$(FEATURES_USED) $(USEMODULE)))
```

Comment:
Doesn't this mean we will *always* use `picolibc` now?

I mean I don't object, but this should probably be discussed in another PR :wink:

Reply:
Wait, I have it reversed: if `picolibc` is in `$(FEATURES_USED)` or `$(USEMODULE)`, `picolibc` is added to `USEMODULE` - how does this make a difference? :exploding_head:

Reply:
It means that if picolibc is already in USEMODULE then don't use newlib. When using FEATURES_REQUIRED in the board Makefile.dep, picolibc is not yet added to FEATURES_USED at this stage so newlib is added the first time this Makefile is parsed.
It doesn't mean picolibc is always used.

Reply:
The other solution is to skip this whole block if picolibc is already in USEMODULE but I'm not sure that makes sense.

Reply:
Ah now it's dawning upon me: `FEATURES_REQUIRED` gets evaluated to late, that's why first the `newlib` path is taken.

That`s why you changed it to

    USEMODULE += picolibc

Reply:
It took me a bit of time to figure that out :)

Reply:
Can you also add it to `cpu/fe310` for consistency.
I hope Kconfig will soon redeem us from this (*right?* *right?*).

Reply:
> Can you also add it to cpu/fe310 for consistency.

I'm not sure, it's very unlikely that a board will have to force the use of picolibc in order to work. Once dependency resolution is managed by Kconfig, we won't need this anymore I hope.

Reply:
Then at least add a comment as to why this is needed - you can squash directly.